tests/testthat/test-class-spec.R

```diff
@@ -0,0 +1,141 @@
+test_that("can work with R7 classes", {
+  klass <- new_class("klass")
+  expect_equal(as_class(klass), klass)
+
+  expect_equal(class_type(klass), "r7")
+  expect_equal(class_desc(klass), "<klass>")
+  expect_equal(class_deparse(klass), "klass")
+
+  obj <- klass()
+  expect_equal(obj_type(obj), "r7")
+  expect_equal(obj_desc(obj), "<klass>")
+  expect_equal(class_inherits(obj, klass), TRUE)
+})
+
+test_that("can work with unions", {
+  klass <- new_union(text, number)
+  expect_equal(as_class(klass), klass)
+
+  expect_equal(class_type(klass), "r7_union")
+  expect_equal(class_desc(klass), "<text> u <number>")
+  expect_equal(class_deparse(klass), "new_union(text, number)")
+
+  # Can't have an instance of a union
+  expect_equal(class_inherits(text("x"), klass), TRUE)
+  expect_equal(class_inherits(number(1), klass), TRUE)
```

Comment:
What does this comment mean?

Reply:
It's a placeholder for the missing pattern from the other tests. I'll clarify.